src/variant_detection/snp_indel_detection.cpp

```diff
@@ -0,0 +1,94 @@
+#include <iostream>
+
+#include <seqan3/alphabet/cigar/cigar.hpp>
+#include <seqan3/core/debug_stream.hpp>
+#include <seqan3/io/sam_file/input.hpp>
+
+#include "iGenVar.hpp"
+#include "variant_detection/bam_functions.hpp"          // check the sam flags
+#include "variant_detection/snp_indel_detection.hpp"    // detect_snp_and_indel
+
+void detect_snp_and_indel(std::filesystem::path const & reads_filename)
+{
+    // Get the header information and set the necessary fields.
+    using sam_fields = seqan3::fields<seqan3::field::flag,       // 2: FLAG
+                                      seqan3::field::ref_id,     // 3: RNAME
+                                      seqan3::field::ref_offset, // 4: POS
+                                      seqan3::field::mapq,       // 5: MAPQ
+                                      seqan3::field::cigar>;     // 6: CIGAR
+
+    seqan3::sam_file_input reads_file{reads_filename, sam_fields{}};
+    auto const & header = reads_file.header();
+
+    // Check that the file is sorted.
+    if (header.sorting != "coordinate")
+        throw seqan3::format_error{"ERROR: Input file must be sorted by coordinate (e.g. samtools sort)"};
+
+    // Prepare one activity vector per reference sequence.
```

Comment:
Question: Are we talking about different references that were aligned to (with different sam files), or just individual fasta files for e.g. the individual chromosomes?

Reply:
Aaah I think its the second one, as you use ref_id below. :)

Reply:
In one sam file there can be linked various reference sequences. Whether they come from a single or multiple sequence files, we need to discuss at some point, but here I just use the ref ids from the sam file.